src/main/java/universe/UniverseTypeUtil.java

```diff
@@ -110,12 +110,13 @@ public static boolean inStaticScope(TreePath treePath) {
         return in;
     }
 
-    public static void defaultConstructorReturnToSelf(Element elt, AnnotatedTypeMirror type) {
+    public static void defaultConstructorReturnToSelf(
+            UniverseAnnotatedTypeFactory atm, Element elt, AnnotatedTypeMirror type) {
         if (elt.getKind() == ElementKind.CONSTRUCTOR
                 && type instanceof AnnotatedTypeMirror.AnnotatedExecutableType) {
             ((AnnotatedTypeMirror.AnnotatedExecutableType) type)
                     .getReturnType()
-                    .addMissingAnnotations(Arrays.asList(UniverseAnnotationMirrorHolder.SELF));
+                    .addMissingAnnotations(Arrays.asList(atm.SELF));
```

Comment:
Please rename `atm` to `realTypeFactory` or something else more suitable (all other uses of `atm` in this file are for `AnnotatedTypeMirror`, with some uses confusingly being named `type`).